src/token/signature.c

```diff
@@ -34,32 +38,64 @@ size_t init_token_signature_message(uint8_t *secret, token_t *token, uint8_t *ou
 }
 
 // sign token
-void sign_token(uint8_t *secret, token_t *token, uint8_t *signature) {
+int sign_token(uint8_t secret[static SECRET_LEN],
+               token_t *token,
+               uint8_t *signature,
+               size_t siglen) {
     uint8_t message[MAX_MESSAGE_LEN];
-    size_t message_len = init_token_signature_message(secret, token, message);
+
+    if (secret == NULL || token == NULL || signature == NULL || siglen < 32) {
+        return 1;
+    }
+
+    size_t message_len = init_token_signature_message(secret, token, message, MAX_MESSAGE_LEN);
+    if (!message_len) {
+        return 2;
+    }
     cx_hash_sha256(message, message_len, signature, 32);
+    return 0;
 }
 
 // verify token signature
-bool verify_token_signature(uint8_t *secret, token_t *token, uint8_t *signature) {
+int verify_token_signature(uint8_t *secret, token_t *token, uint8_t *signature, size_t siglen) {
```

Comment:
Should we validate if `secret`, `token`, or `signature` are `NULL`?

Reply:
They are verified on the sign_token below, but I'll add the verification here as well